### PR TITLE
Issue #13347: False Positive and Negative in SummaryJavaDocCheck

### PR DESCRIPTION
**Pull request by AayushSaini101**
Related: https://github.com/checkstyle/checkstyle/issues/13347
Diff Regression config: https://gist.githubusercontent.com/AayushSaini101/2db49b2556a5ed7f2df81b3624661c38/raw/d9e6dac89729ca2436edd6506fe76dcbe3feb0cd/gistfile1.txt

### CODE REVIEW

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadoc1.java [hunk 1]
```diff
@@ -13,7 +13,7 @@ public class InputSummaryJavadoc1 {
     // violation 2 lines below 'Forbidden summary fragment'
     /**
      * {@summary Returns the customer ID.
-     *  This method returns. }
+     *  This method returns.}
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadoc1.java:16]: We almost never change inputs, the more they weird the better.
Please change list of violations.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java [hunk 2]
```diff
@@ -467,9 +467,9 @@ private static String getVisibleContent(String summary) {
      * @param firstSentence string with first sentence.
      * @return {@code true} if first sentence contains forbidden summary fragment.
      */
-    private boolean containsForbiddenFragment(String firstSentence) {
+    public boolean containsForbiddenFragment(String firstSentence) {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java:470]: Revert

**Review by romani — CHANGES_REQUESTED** [review]
Items:

**Review by nrmancuso — COMMENTED** [review]
Please add all test cases with matching configs from issue description, also generate a regression report via GitHub action. See https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#executing-generation.

**Comment by romani** [comment]
@AayushSaini101 , as you switched to more complicated causes, please reply to each review items as "done" to make sure you addressed them. It helps a lot to speedup review.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadoc1.java (hunk 1 above)
Comment (AayushSaini101) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadoc1.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java (hunk 2 above)
Comment (AayushSaini101) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java]: Done

**Comment by AayushSaini101** [comment]
> @AayushSaini101 , as you switched to more complicated causes, please reply to each review items as "done" to make sure you addressed them. It helps a lot to speedup review.

Thanks for the instructions

**Comment by AayushSaini101** [comment]
GitHub, generate report

**Comment by AayushSaini101** [comment]
> Please add all test cases with matching configs from issue description, also generate a regression report via GitHub action. See https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#executing-generation.

Done new test case added

**Comment by AayushSaini101** [comment]
GitHub, generate report

**Comment by AayushSaini101** [comment]
GitHub, generate report

**Comment by AayushSaini101** [comment]
> Please add all test cases with matching configs from issue description, also generate a regression report via GitHub action. See https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#executing-generation.

Done

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java [hunk 3]
```diff
@@ -306,12 +306,20 @@ public void testForbidden2() throws Exception {
...
     @Test
     public void testSummaryJavaDoc() throws Exception {
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+
+        verifyWithInlineConfigParser(
+                getPath("InputSummaryJavadoc1.java"), expected);
+    }
+
+    @Test
+    public void testSummaryJavaDoc1() throws Exception {
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java:316]: `testSummaryJavaDoc2()`

as you use InputSummaryJavadoc2.java

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadoc2.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+forbiddenSummaryFragments = ^Returns the customer ID. This method returns. $
+period = .
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.summaryjavadoc;
+
+public class InputSummaryJavadoc2 {
+
+    // violation 2 lines below 'Forbidden summary fragment'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadoc2.java:13]: why two violations?

you have in test method:
```
final String[] expected = {
            "15: " + getCheckMessage(MSG_SUMMARY_JAVADOC),
        };
```

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java (hunk 3 above)
Comment (AayushSaini101) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java:316]: Thanks, Changing the name

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadoc2.java (hunk 4 above)
Comment (AayushSaini101) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadoc2.java:13]: There are two lines that are matching with the forbiddenSummaryFragments

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadoc2.java:13]: but in array we have only one violation.
and in inlined comment you mentioned that there are two.

if I am blind ..... ok
please share CLI output for this Input.
here is hoow to build jar of your changes - https://github.com/checkstyle/checkstyle/wiki/How-to-run-certain-phases-and-validations#how-to-generate-all-binaries-and--alljar--too

Comment (AayushSaini101) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadoc2.java:13]: index.java 
```
public class index {
    /**
     * {@summary Returns the customer ID.
     *  This method returns. }
     */
    public void foo() {
    }
}
```
config.xml
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC 
 "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
 "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="SummaryJavadoc">
            <property name="violateExecutionOnNonTightHtml" value="true" />
            <property name="forbiddenSummaryFragments"
                value="^Returns the customer ID. This method returns. $" />
                            <!-- attention to extra space before $  ^^^ -->
        </module>

    </module>
</module>
```
Output: 
```
aaayush@aaayush-mac GFG % java -jar checkstyle-10.12.5-SNAPSHOT-all.jar -c config.xml index.java
Starting audit...
[ERROR] /Users/aaayush/Desktop/GFG/index.java:3: Forbidden summary fragment. [SummaryJavadoc]
Audit done.
Checkstyle ends with 1 errors
```
@romani This is the status of the new build jar

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadoc2.java:13]: My bad , I blindly read it as 2 violations, but it is violation 2 lines below

**Review by romani — APPROVED** [review]
Ok to merge

**Review by nrmancuso — APPROVED** [review]
Sonar is failing, @AayushSaini101 did we solve this issue?

**Pull request merged**